sql/src/main/java/io/crate/expression/symbol/FunctionCopyVisitor.java

```diff
@@ -79,21 +89,35 @@ private Function twoArgs(Function func, C context) {
         Symbol arg2 = func.arguments().get(1);
         Symbol newArg2 = requireNonNull(process(arg2, context), "function arguments must never be NULL");
 
-        if (arg1 == newArg1 && arg2 == newArg2) {
+        Symbol filter = func.filter();
+        Symbol newFilter = processSafe(filter, context);
+
+        if (arg1 == newArg1 && arg2 == newArg2 && filter == newFilter) {
             return func;
         }
-        return new Function(func.info(), List.of(newArg1, newArg2), func.filter());
+        return new Function(func.info(), List.of(newArg1, newArg2), newFilter);
     }
 
     private Function oneArg(Function func, C context) {
         assert func.arguments().size() == 1 : "size of arguments must be one";
         Symbol arg = func.arguments().get(0);
         Symbol newArg = requireNonNull(process(arg, context), "function arguments must never be NULL");
 
-        if (arg == newArg) {
+        Symbol filter = func.filter();
+        Symbol newFilter = processSafe(filter, context);
+
+        if (arg == newArg && filter == newFilter) {
             return func;
         }
-        return new Function(func.info(), List.of(newArg), func.filter());
+        return new Function(func.info(), List.of(newArg), newFilter);
+    }
+
+    @Nullable
+    private Symbol processSafe(@Nullable Symbol symbol, C context) {
+        if (symbol != null) {
+            return process(symbol, context);
```

Comment:
```suggestion
            return symbol.accept(this, context);
```

to avoid 1 extra function call.